netket/machine/jax.py

```diff
@@ -347,6 +347,160 @@ def JaxRbm(hilbert, alpha, dtype=complex):
     )
 
 
+def MPSPeriodic(hilbert, bond_dim, diag=False, symperiod=None, dtype=complex):
+    r"""
+    Constructs a periodic Matrix Product State (MPS) for a quantum state of discrete
+    degrees of freedom, wrapped as Jax machine.  The MPS is defined as
+
+    .. math:: \Psi(s_1,\dots s_N) = \Tr(A[s_1]\dots A[s_N]),
+
+    for arbitrary local quantum numbers :math:`s_i`, where :math:`A[s_1]` is a matrix
+    of dimension (bdim,bdim), depending on the value of the local quantum number :math:`s_i`.
+
+        Args:
+            hilbert: Hilbert space on which the state is defined. Should be a
+                subclass of `netket.hilbert.Hilbert`.
+            bond_dim (int): Virtual dimension of the MPS tensors.
+            diag (bool): Whether or not to use diagonal matrices in the MPS tensors.
+                default=False
+            symperiod (int): Periodicity in the chain of MPS tensors. The chain of
+                MPS tensors is constructed as a sequence of identical unit cells
+                consisting of symperiod tensors. if None, symperiod equals the
+                number of physical degrees of freedom.
+                default=None
+            dtype: complex or float, whether the variational parameters of the MPS
+                are real or complex. default=complex
+
+        returns:
+            Jax machine of the Matrix Product state.
+    """
+    return Jax(
+        hilbert,
+        stax.serial(MpsPeriodicLayer(hilbert, bond_dim, diag, symperiod, dtype)),
```

Comment:
I think we don't need `stax.serial`. `MpsPeriodicLayer` is effectively a network by itself.

Reply:
yes but our ```machine.Jax``` only takes stax modules?

Reply:
I don't think so.
A stax thing is simply a function returning two functions. see https://github.com/google/jax/blob/7c687b245b34397c13563a714ad9bf0290b419e3/jax/experimental/stax.py#L279 

so MpsPeriodicLayer respects this.

But then, maybe I'm wrong...